x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/lib/search_fields.tsx

```diff
@@ -0,0 +1,227 @@
+  }
+
+  if (metadata.matchType) {
+    score += 5;
+  }
+
+  if (metadata.fullyMatchType) {
+    score += 5;
+  }
+
+  return score;
+};
+
+const getJSXdisplayFromMeta = (
+  searchData: SearchData,
+  fieldData: FieldData,
+  metadata: Omit<SearchMetadata, 'score' | 'display'>
+): JSX.Element => {
+  const { term } = searchData;
+  const { path } = fieldData;
+
+  let display: JSX.Element = <span>{path}</span>;
+
+  if (metadata.fullyMatchPath) {
+    display = (
+      <span>
+        <strong>{path}</strong>
+      </span>
+    );
+  } else if (metadata.matchStartOfPath) {
+    const endString = path.substr(term.length, path.length);
+    display = (
+      <span>
+        <strong>{term}</strong>
+        {endString}
+      </span>
+    );
+  } else if (metadata.matchPath) {
+    const { stringMatch } = metadata;
+    const charIndex = path.toLowerCase().indexOf(stringMatch!.toLowerCase());
+    const startString = path.substr(0, charIndex);
+    const endString = path.substr(charIndex + stringMatch!.length);
+    display = (
+      <span>
+        {startString}
+        <strong>{stringMatch}</strong>
+        {endString}
+      </span>
+    );
+  }
+
+  return display;
+};
+
+const getSearchMetadata = (searchData: SearchData, fieldData: FieldData): SearchMetadata => {
+  const { term, type, searchRegexArray } = searchData;
+  const typeToCompare = type ?? term;
+
+  const matchStartOfPath = fieldData.path.startsWith(term);
+  const fullyMatchPath = term === fieldData.path;
+  const matchType = fieldData.type.includes(typeToCompare);
+  const fullyMatchType = typeToCompare === fieldData.type;
+
+  let stringMatch: string | null = null;
+
+  if (fullyMatchPath) {
+    stringMatch = fieldData.path;
+  } else {
+    // Execute all the regEx and sort them with the one that has the most
+    // characters match first.
+    const arrayMatch = searchRegexArray
+      .map(regex => regex.exec(fieldData.path))
+      .filter(Boolean)
+      .sort((a, b) => b![0].length - a![0].length);
+
+    if (arrayMatch.length) {
+      stringMatch = arrayMatch[0]![0];
+    }
+  }
+
+  const matchPath = stringMatch !== null;
+
+  const metadata = {
+    matchPath,
+    matchStartOfPath,
+    fullyMatchPath,
+    matchType,
+    fullyMatchType,
+    stringMatch,
+  };
+
+  const score = calculateScore(metadata);
+  const display = getJSXdisplayFromMeta(searchData, fieldData, metadata);
+
+  return {
+    ...metadata,
+    display,
+    score,
+  };
+};
+
+/**
+ * Return an array of array combining sibling elements
+ * In: ['A', 'B', 'C', 'D']
+ * Out: [['A', 'B'], ['B', 'C'], ['C', 'D']]
+ *
+ * @param arr Array of string
+ */
+const getSubArrays = (arr: string[]): string[][] => {
+  let i = 0;
+  const result = [];
+  while (i < arr.length - 1) {
+    result.push([arr[i], arr[i + 1]]);
+    i++;
+  }
+  return result;
+};
+
+const getRegexFromArray = (array: string[]): RegExp[] => {
+  const termsRegexArray = array.map(value => new RegExp(value, 'i'));
```

Comment:
We may want to consider applying something like https://www.npmjs.com/package/escape-string-regexp here

Reply:
Cool, I'll keep it in mind if we start having more regex string to escape. 👍  For now, we only escape 2 chars in the file so it might be overkill.